### PR TITLE
Porting to ros2

### PR DESCRIPTION
**Pull request by destogl**
Hi maintainers,

here the port of the repo ros2. We need those for ROS2 Driver and GPIOController added in PickNikRobotics/Universal_Robots_ROS2_Driver#42.

Could you create a branch for ros2? If you also need help with CI setup (GH-actions) I can crate a follow-up PR for it.

Thanks!

### CODE REVIEW

**Comment by destogl** [comment]
Oh, sorry for CHANGELOG delete I didn't see that is missing from the first PR. Should I "re-add" it? It's a new ROS version so not sure.

**Comment by destogl** [comment]
Ping: @gavanderhoorn and @fmauch - how should we proceed on this?

**Comment by fmauch** [comment]
Disclaimer: Not being a maintainer of this repo, my opinion is only an opinion of a developer using this repo.

Merging those changes into melodic-devel seems not to make very much sense, as the PR completely ignores ROS1 / catkin if I understand this correctly.

It would probably make sense to add a separate `foxy-devel` branch?

Personally I think it would make very much sense to include that into this repository to keep `ur_msgs` at one place for both, ROS1 and ROS2.

**Comment by destogl** [comment]
@gavanderhoorn are you a maintainer of this repo? If so, can you please add a new branch for ROS 2. I can also add CI and other organizational details to it...

**Comment by urrsk** [comment]
@gavanderhoorn What about making a ROS2 branch?

**Comment by fmauch** [comment]
@destogl I've added some changes to your branch to include CI

We'll figure out a strategy to manage this in the long run, This should happen in the next couple of weeks.

File: package.xml [hunk 1]
```diff
@@ -1,35 +1,30 @@
...
-      <tags>
-        <tag>messages</tag>
-        <tag>ros-industrial</tag>
-        <tag>services</tag>
-        <tag>universal_robot</tag>
-      </tags>
-    </rosindex>
-    <ros_industrial>
-      <support_level value="community" />
-    </ros_industrial>
```
Comment (gavanderhoorn) [inline package.xml]: please retain these elements.

**Review by gavanderhoorn — CHANGES_REQUESTED** [review]

File: package.xml (hunk 1 above)
Comment (fmauch) [reply package.xml]: done in d6449d7

**Comment by fmauch** [comment]
I've created a foxy-devel branch and changed the target of this branch to that one.

File: package.xml [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
+
+  <exec_depend>rosidl_default_runtime</exec_depend>
+
+  <test_depend>ament_lint_auto</test_depend>
+  <test_depend>ament_lint_common</test_depend>
+
+  <member_of_group>rosidl_interface_packages</member_of_group>
 
   <export>
     <architecture_independent />
```
Comment (gavanderhoorn) [inline package.xml]: Are ROS 2 msg pkgs still `architecture_independent`? They're not header-only any more, so I would not expect that to still be the case.

Comment (fmauch) [reply package.xml]: Yes, thanks for spotting that. Fixed in 7ea5b71

**Comment by fmauch** [comment]
@gavanderhoorn As I am co-author of this and you've requested changes in the past, could you please review this once more?

**Review by gavanderhoorn — APPROVED** [review]

**Pull request merged**